src/test/scala/chiffreTests/passes/FaultInstrumentationSpec.scala

```diff
@@ -58,7 +67,115 @@ class FaultInstrumentationSpec extends ChiselFlatSpec {
+    
+    val circuit = Parser.parse(input)
+    val state = CircuitState(circuit, MidForm, Seq.empty, None)
+    val connections = new ArrayBuffer[Connect]()
+    f.execute(state).circuit.modules.filter(_.name == component.module.name).foreach(_.mapStmt(collect(connections)))
+    connections.map(_.serialize) should contain ("x_fault.a.a1 <= asUInt(bits(IdentityInjector.io.out, 6, 6))")
+    connections.map(_.serialize) should contain ("x_fault.a.a2 <= asUInt(bits(IdentityInjector.io.out, 5, 3))")
+    connections.map(_.serialize) should contain ("x_fault.b <= asUInt(bits(IdentityInjector.io.out, 2, 0))")
+    connections.last.expr.serialize should be ("cat(cat(asUInt(x.a.a1), asUInt(x.a.a2)), asUInt(x.b))")
+  }
+
+  it should "inject faults into a combined vector and bundle type" in {
+    val component = ComponentName("x", ModuleName("top", CircuitName("top")))
+    val compMap = Map(component.module.name -> Seq((component, "dummyId", classOf[IdentityInjector])))
+    val f = new FaultInstrumentation(compMap)
+
+    val tpe = "{a: {a1: UInt<1>, a2: UInt<1>[2]}, b: UInt<3>}[2]"
+    val input = s"""|circuit top:
+                    |  module top:
+                    |    input clock: Clock
+                    |    input in: $tpe
+                    |    output out: UInt<1>
+                    |    reg x: $tpe, Clock
+                    |    x <= in
+                    |    out <= orr(asUInt(x))
+                    |""".stripMargin
+    
+    val circuit = Parser.parse(input)
+    val state = CircuitState(circuit, MidForm, Seq.empty, None)
+    val connections = new ArrayBuffer[Connect]()
+    f.execute(state).circuit.modules.filter(_.name == component.module.name).foreach(_.mapStmt(collect(connections)))
+    connections.map(_.serialize) should contain ("x_fault[0].b <= asUInt(bits(IdentityInjector.io.out, 2, 0))")
+    connections.map(_.serialize) should contain ("x_fault[0].a.a2[0] <= asUInt(bits(IdentityInjector.io.out, 3, 3))")
+    connections.map(_.serialize) should contain ("x_fault[0].a.a2[1] <= asUInt(bits(IdentityInjector.io.out, 4, 4))")
+    connections.map(_.serialize) should contain ("x_fault[0].a.a1 <= asUInt(bits(IdentityInjector.io.out, 5, 5))")
+    connections.map(_.serialize) should contain ("x_fault[1].b <= asUInt(bits(IdentityInjector.io.out, 8, 6))")
+    connections.map(_.serialize) should contain ("x_fault[1].a.a2[0] <= asUInt(bits(IdentityInjector.io.out, 9, 9))")
+    connections.map(_.serialize) should contain ("x_fault[1].a.a2[1] <= asUInt(bits(IdentityInjector.io.out, 10, 10))")
+    connections.map(_.serialize) should contain ("x_fault[1].a.a1 <= asUInt(bits(IdentityInjector.io.out, 11, 11))")
+    connections.last.expr.serialize should be ("cat(cat(cat(asUInt(x[1].a.a1), cat(asUInt(x[1].a.a2[1]), asUInt(x[1].a.a2[0]))), asUInt(x[1].b)), cat(cat(asUInt(x[0].a.a1), cat(asUInt(x[0].a.a2[1]), asUInt(x[0].a.a2[0]))), asUInt(x[0].b)))")
+  }
+
+  it should "error if a 0-width Vec is instrumented" in {
```

Comment:
The error message here from FIRRTL is confusing. I think it would be more informative to the user if we errored on trying to instrument anything which turns out to have a zero width. This should be trivial to check looking at the `bitWidth(t)` and trowing an informative `FaultInstrumentationException`.